api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/AdvancedDiseaseAndTBCascadeCohortQueries.java

```diff
@@ -0,0 +1,1485 @@
+            + "                                            AND o.voided = 0 "
+            + "                                            AND e.encounter_type = ${53} "
+            + "                                            AND o.concept_id = ${6272} "
+            + "                                            AND e.location_id = :location "
+            + "                                            AND o.obs_datetime BETWEEN :startDate AND :endDate "
+            + "                                     GROUP  BY e.patient_id) recent_state ON recent_state.patient_id = e.patient_id "
+            + "                   WHERE  e.voided = 0 "
+            + "                          AND o.voided = 0 "
+            + "                          AND o.concept_id = ${6272} "
+            + "                          AND o.value_coded = ${1705} "
+            + "                          AND e.location_id = :location "
+            + "                          AND o.obs_datetime = recent_state.state_date "
+            + "                   GROUP  BY e.patient_id) reinitiated ON reinitiated.patient_id = p.patient_id "
+            + "WHERE  p.voided = 0 "
+            + "GROUP  BY p.patient_id";
+
+    StringSubstitutor sb = new StringSubstitutor(getMetadata());
+    cd.setQuery(sb.replace(query));
+
+    return cd;
+  }
+
+  /**
+   *
+   *
+   * <ul>
+   *   <li>Absolute CD4 count results (CD4 Absoluto= ANY RESULT) registered in the Ficha Clinica –
+   *       Mastercard during the inclusion periodor
+   *   <li>CD4 count result (CD4 absoluto - Último CD4= ANY RESULT) registered in the Ficha Resumo –
+   *       Mastercard with the result date during the inclusion period or
+   *   <li>Absolute CD4 count results (CD4 Absoluto= ANY RESULT) registered in Laboratory form
+   *       during the inclusion period or
+   *   <li>Absolute CD4 count results (CD4 Absoluto= ANY RESULT) registered in the Electronic Lab
+   *       form during the inclusion period or
+   *   <li>Absolute CD4 count results (CD4 Absoluto= ANY RESULT) registered in the Ficha DAH with
+   *       the result date during the inclusion period
+   * </ul>
+   *
+   * <p>Note: For clients with more than one CD4 count result registered within this period, the
+   * most recent result will be considered
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getPatientsWithCD4Count() {
+
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Number of clients with a CD4 count during inclusion period");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "End Date", Location.class));
+
+    String query =
+        "SELECT p.patient_id "
+            + "FROM   patient p "
+            + "       INNER JOIN (SELECT e.patient_id, Max(DATE(e.encounter_datetime)) AS result_date "
```

Comment:
Acho que não precisas deste maximo

Reply:
Will be useful for debugging, as the specification refers to the most recent date.